cloudpickle/cloudpickle.py

```diff
@@ -141,22 +142,38 @@ def _whichmodule(obj, name):
     return None
 
 
-def _is_global(obj, name=None):
+if sys.version_info[:2] < (3, 7):
+    def _get_module_attr(obj):
+        if isinstance(obj, typing.TypeVar):
+            return None
+        return getattr(obj, '__module__', None)
+else:
+    def _get_module_attr(obj):
+        return getattr(obj, '__module__', None)
+
+
+def _is_importable_by_name(obj, name=None):
     """Determine if obj can be pickled as attribute of a file-backed module"""
+    return _lookup_module_and_qualname(obj, name) is not None
+
+
+def _lookup_module_and_qualname(obj, name=None):
     if name is None:
-        name = getattr(obj, '__qualname__', None)
-    if name is None:
-        name = getattr(obj, '__name__', None)
+        if isinstance(obj, typing.TypeVar):
```

Comment:
I'd rather have this logic inside `save_typevar` , and use the `name` kwarg of `_lookup_module_and_qualname`.

Reply:
Depends what the desired behaviour of this function is when `name=None` is passed. Is it:

1. Get the name of the object, or
2. Get the `__qualname__` attribute of the object?

Reply:
We should leave the behavior of `_lookup_module_and_qualname` as it was, and simply use `_lookup_module_and_qualname(typevar, name=typevar.__name__)` in `save_typevar`.

Reply:
Note, this change was in response to https://github.com/cloudpipe/cloudpickle/pull/350#discussion_r391470319.

Reply:
I understand, but this does not conflict with my point. Could you precise what problem you see in my suggestion?

Reply:
I think that with my suggestion, we could drop relying on  `__name__` in `_lookup_module_in_qualname`, as https://github.com/cloudpipe/cloudpickle/pull/350#discussion_r391639128 suggest the `__name__` branch is legacy `Python 2` handling.

Reply:
I agree with the suggestion to do `_lookup_module_and_qualname(typevar, name=typevar.__name__) in save_typevar`.

Then we could drop the `if name is None: name = getattr(obj, '__name__', None)` case but I think we should keep it for now explaining in an inline comment that this should never be used and that we will probably remove it in the future unless someone can find a case for this.

Reply:
Yes I agree, I'm fine with leaving this code + adding an inline comment.

Reply:
I agree. I'm fine with leaving the `__name__` branch + adding an inline comment.

Reply:
Just to confirm that I understood, the changes are:

- revert the beginning of `_lookup_module_and_qualname` to:

```
if name is None:
	name = getattr(obj, '__qualname__', None)
if name is None:
	name = getattr(obj, '__name__', None)
```

- in every place where `_lookup_module_and_qualname` is called with an `obj` that is statically known to be a `TypeVar`, pass `name=obj.__name__`; that is, in `save_typevar` and the three `test_lookup_module_and_qualname` tests

Reply:
Yes.